CONTRIBUTING.md

```diff
@@ -1,9 +1,47 @@
 # How to contribute to Aleph
 
-This document is a work in progress!
+Thanks for helping out! Aleph is a collaborative project that wouldn’t exist without the generous time and effort of many people.
+
+This document is a work in progress, so if there's anything you feel needs to be added, pleae let us know, or file a PR.
+
+## Local development environment
+
+We require a recent version of [Leiningen](https://leiningen.org/), and a minimum Java version of 8. Running the deps update script will require bash on your system. Other than that, we have no specific requirements.
+
+## Testing
+
+`lein test` should be run, and pass, before pushing up to GitHub.
+
+If you are fixing a bug, create a test that demonstrates it _first_, then write the fix. This ensures our understanding of the problem is correct.
+
+## Linting
+
+[clj-kondo](https://github.com/clj-kondo/clj-kondo) is recommended for linting. Because of the potemkin library and various proto-potemkin macros, Aleph can trigger a lot of false positives, and clj-kondo is the least bad at understanding them. (Handling this is ongoing; PRs welcome.) 
 
 ## Updating dependencies
 
-After you've made your changes to `project.clj`, be sure to run the tests with `lein test`!
+There’s some extra steps to test when updating dependencies:
+
+1. Obviously, after you've made your changes to `project.clj`, be sure to rerun the tests with `lein test`.
```

Comment:
```suggestion
1. After you've made your changes to `project.clj`, be sure to rerun the tests with `lein test`.
```
I prefer not to use phrases like "just", "simply" or "obviously" in documentation. When I'm looking for information I couldn't figure out myself and run into a phrase like that it feels very condescending.